ddl/table.go

```diff
@@ -96,7 +97,13 @@ func onCreateView(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error)
 		// none -> public
 		tbInfo.State = model.StatePublic
 		tbInfo.UpdateTS = t.StartTS
-		err = t.CreateTable(schemaID, tbInfo)
+		if oldTbInfoID > 0 && orReplace {
+			err = t.DropTableOrView(schemaID, oldTbInfoID, true)
+			if err != nil {
```

Comment:
Should we set `job.State = model.JobStateCancelled` here?

Reply:
I think don't need to cancel job here, we can retry this job.

Reply:
@XuHuaiyu Please take a look at my previous comment.

Reply:
OK

Reply:
@winkyao @crazycs520 give us a situation when `DropTable` fail loop....Do you have any idea?